src/charm.py

```diff
@@ -66,12 +66,12 @@ def version(self) -> Optional[str]:
 
 
 class AlertmanagerCharm(CharmBase):
-    _container_name: str = "alertmanager"  # automatically determined from charm name
+    _container_name = "alertmanager"  # automatically determined from charm name
     _layer_name = "alertmanager"  # layer label argument for container.add_layer
-    _service_name: str = "alertmanager"  # chosen arbitrarily to match charm name
-    _peer_relation_name: str = "replicas"  # must match metadata.yaml peer role name
-    _api_port: int = 9093  # port to listen on for the web interface and API
-    _ha_port: int = 9094  # port for HA-communication between multiple instances of alertmanager
+    _service_name = "alertmanager"  # chosen arbitrarily to match charm name
+    _peer_relation_name = "replicas"  # must match metadata.yaml peer role name
+    _api_port = 9093  # port to listen on for the web interface and API
+    _ha_port = 9094  # port for HA-communication between multiple instances of alertmanager
```

Comment:
If these are not configurable, perhaps better as constants for now?

Reply:
do you mean literals? I made this to avoid duplicating literals

Reply:
unless you meant module-wide variables in all-caps? If this is debatable then I'd argue for scope correctness.

Reply:
It's definitely more common python practice to put them at the module level. The caps, however, I agree is unnecessary.